modules/java/generator/src/cpp/Mat.cpp

```diff
@@ -1845,26 +1868,19 @@ template<typename T> static int mat_put(cv::Mat* m, int row, int col, int count,
     return res;
 }
 
-
-extern "C" {
-
-JNIEXPORT jint JNICALL Java_org_opencv_core_Mat_nPutB
-    (JNIEnv* env, jclass, jlong self, jint row, jint col, jint count, jbyteArray vals);
-
-JNIEXPORT jint JNICALL Java_org_opencv_core_Mat_nPutB
-    (JNIEnv* env, jclass, jlong self, jint row, jint col, jint count, jbyteArray vals)
+template<class ARRAY> static jint java_mat_put(JNIEnv* env, jclass, jlong self, jint row, jint col, jint count, ARRAY vals)
```

Comment:
I suggest removal of the unused `jclass` param and dropping it at the function call.